packages/app-storage/src/Selection/Modules.tsx

```diff
@@ -101,11 +110,18 @@ class Modules extends React.PureComponent<Props, State> {
 
   private onAdd = (): void => {
     const { onAdd } = this.props;
-    const { key, values } = this.state;
+    const { key, values, params } = this.state;
+    let newParams: Array<MethodParams> = new Array();
+
+    // if the method has a parameter, store the value and
+    //  type of the method's parameter in the same object
+    if (values.length && params.length) {
+      newParams.push(Object.assign({}, values[0], params[0]['type']));
+    }
 
     onAdd({
       key,
-      params: values
+      params: newParams
```

Comment:
It would seem like all this bubbles up now, as opposed to just the `isValid` & `value`?